libvuln/driver/updaterset.go

```diff
@@ -0,0 +1,87 @@
+package driver
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// ErrExists is an error returned if the updater
+// already exists in the set.
+type ErrExists struct {
+	Updater []string
+}
+
+func (e ErrExists) Error() string {
+	return fmt.Sprintf("%v", e.Updater)
+}
+
+// UpdaterSet holds a deduped
+// set of updaters
+type UpdaterSet struct {
+	Set map[string]Updater
```

Comment:
If users are meant to always use Add and Merge to manipulate the set, then this shouldn't be exported.